backend/src/main/java/ca/gov/dtsstn/cdcp/api/web/v1/controller/ConfirmationCodeController.java

```diff
@@ -0,0 +1,48 @@
+package ca.gov.dtsstn.cdcp.api.web.v1.controller;
+import java.util.List;
+import java.util.stream.StreamSupport;
+
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import ca.gov.dtsstn.cdcp.api.service.ConfirmationCodeService;
+import ca.gov.dtsstn.cdcp.api.web.v1.model.ConfirmationCodeModel;
+import ca.gov.dtsstn.cdcp.api.web.v1.model.mapper.ConfirmationCodeModelAssembler;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.Parameter;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.constraints.NotBlank;
+import ca.gov.dtsstn.cdcp.api.service.domain.ConfirmationCode;
+
+@Validated
+@RestController
+@RequestMapping({ "/api/v1" })
+@Tag(name = "confirmationCodes", description = "CRUD endpoint for confirmation codes.")
+public class ConfirmationCodeController {
+    
+    private final ConfirmationCodeModelAssembler confirmationCodeModelAssembler;
+    private final ConfirmationCodeService confirmationCodeService;
+
+    public ConfirmationCodeController(ConfirmationCodeModelAssembler confirmationCodeModelAssembler, ConfirmationCodeService confirmationCodeService){
+        this.confirmationCodeModelAssembler = confirmationCodeModelAssembler;
+        this.confirmationCodeService = confirmationCodeService;
+    }
+
+    @GetMapping({ "/users/{userEmail}/confirmationCodes" })
+	@Operation(summary = "Get all confirmation codes for a user.", operationId = "get-confirmation-codes")
+    public List<ConfirmationCodeModel> getConfirmationCodesByUserEmail(
```

Comment:
I'm not sure this is a useful endpoint. When would we ever need to fetch all confirmation codes for a user?

Generally we would email the user a confirmation code, and then the user would give that confirmation code back to us.. this would probably be done by POSTing to an email-validations endpoint.

Reply:
after the chat we had, if a user had requested many codes and did not received all of them at their email inbox, they should still be able to validate themselves with one of the code generated no?

Reply:
Yes.. however no consumer should ever need to request all of the confirmation codes that we have for a user. The consumer will instead `POST /users/{userId}/email-validations` with the confirmation code provided by the user, and the API will either accept or reject that request.